lwt/cohttp_lwt.ml

```diff
@@ -298,7 +298,14 @@ module Make_server(IO:IO)
     Filename.concat docroot rel_path
 
   let respond ?headers ?(flush=true) ~status ~body () =
-    let encoding = Cohttp_lwt_body.transfer_encoding body in
```

Comment:
It's a little more work but you're right that it's better from a rigor perspective.

I did originally think that this logic could be reused between async/lwt. Not the case however.